libsolidity/codegen/CompilerContext.cpp

```diff
@@ -334,14 +334,7 @@ CompilerContext& CompilerContext::appendJump(evmasm::AssemblyItem::JumpType _jum
 
 CompilerContext& CompilerContext::appendPanic(util::PanicCode _code)
 {
-	Whiskers templ(R"({
-		mstore(0, <selector>)
-		mstore(4, <code>)
-		revert(0, 0x24)
-	})");
-	templ("selector", util::selectorFromSignature("Panic(uint256)").str());
-	templ("code", toCompactHexWithPrefix(static_cast<unsigned>(_code)));
-	appendInlineAssembly(templ.render());
+	callYulFunction(utilFunctions().panicFunction(_code), 0, 0);
```

Comment:
The generated code made me also question why do we generate a panic function with a given code, and why not a panic function which takes the code as an argument?

Reply:
I am not sure about this. Does it help optimizer to have function that doesn't have parameters?

Reply:
Even this change helps in the sense that instead of always inlining, only a jump is inserted. But I think it could be improved, at least for the legacy codegen, to insert a call to`panic(code)` instead.